### PR TITLE
Fixed token exposure in `JobResult` traceback and result output when `GitRepositorySync` job fails.

### PR DESCRIPTION
**Pull request by jathanism**
# Closes: #4673 
# What's Changed

- Updated `NautobotDatabaseBackend._get_extended_properties` to always sanitize the traceback output if it is set.
- Overloaded `NautobotDatabaseBackend.prepare_exception()` to always sanitize the CLI error output message when a exception's message payload is a list or tuple. The output from this method is what gets stored as the `result` of the job in such error cases.
 

## Example fetch
Cursor_and_Notification_Center

## Example clone
Cursor_and_Notification_Center

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: changes/4673.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed token exposure in `JobResult` traceback and result output when a `GitRepositorySync` job fails.
```
Comment (glennmatthews) [inline changes/4673.fixed:1]: Perhaps?

```suggestion
Fixed token exposure in `JobResult` traceback and result output when a `GitRepositorySync` job fails in certain ways.
```

Comment (glennmatthews) [reply changes/4673.fixed:1]: Should this be a `.security` item?

File: nautobot/core/celery/backends.py [hunk 2]
```diff
@@ -64,3 +69,16 @@ def _get_extended_properties(self, request, traceback):
...
+    def prepare_exception(self, exc, serializer=None):
+        """Overload default to explicitly sanitize the traceback message result."""
+        exc_info = super().prepare_exception(exc, serializer=serializer)
+
+        # If this is a GitCommandError, attempt to sanitize the message error.
+        # e.g. b"fatal: could not read Password for 'https://abc123@github.com': terminal prompts disabled"
+        if exc_info["exc_type"] == "GitCommandError":
+            exc_msg = list(exc_info["exc_message"])  # It starts as a tuple.
+            exc_msg[-1] = sanitize(exc_msg[-1].decode("utf-8"))  # The message will be bytes.
+            exc_info["exc_message"] = tuple(exc_msg)  # Turn it back into a tuple.
```
Comment (glennmatthews) [inline nautobot/core/celery/backends.py:82]: This is a lot more narrowly targeted than I'd personally prefer. Can we not just sanitize every part of `exc_message` unconditionally?

File: changes/4673.fixed (hunk 1 above)
Comment (jathanism) [reply changes/4673.fixed:1]: Probably.

File: nautobot/core/celery/backends.py (hunk 2 above)
Comment (jathanism) [reply nautobot/core/celery/backends.py:82]: For the fetch/clone error cases, `exc_message` coming from `GitCommandError` is a list following this pattern:

```
# ([shlex.split(), retval, error_msg])
(["git", "fetch", "other", "args"], 128, b"fatal: could not read Password for 'https://abc123@github.com': terminal prompts disabled"])
```

I was trying `json.loads(sanitize(json.dumps(exc_info["exc_message"])))` to do it in one shot but that stupid error message is `bytes` for some reason.

The other thing is that I have no assertion that other errors would be in this same format, so I kept it super specific to the reported issue.

Comment (jathanism) [reply nautobot/core/celery/backends.py:82]: Also I figure now that there is a pattern for sanitizing exceptions coming from Celery, that we might end up with mappings on how to do each one as-needed, unless we can come-up with a byte-steamrolling-JSON-encoder.

File: changes/4673.security [hunk 3]
```diff
@@ -0,0 +1 @@
+Fixed token exposure in `JobResult` traceback and result output when a `GitRepositorySync` job fails in scertain ways.
```
Comment (glennmatthews) [inline changes/4673.security:1]:
```suggestion
Fixed token exposure in `JobResult` traceback and result output when a `GitRepositorySync` job fails in certain ways.
```

File: nautobot/core/celery/backends.py [hunk 4]
```diff
@@ -64,3 +69,32 @@ def _get_extended_properties(self, request, traceback):
             )
 
         return extended_props
+
+    def prepare_exception(self, exc, serializer=None):
```
Comment (glennmatthews) [inline nautobot/core/celery/backends.py:73]: Anything we can do as a test case for this functionality? I'm thinking one of our test Jobs that intentionally throws an exception?

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**